src/core/text/TextMetrics.js

```diff
@@ -0,0 +1,191 @@
+import Text from './Text';
```

Comment:
@themoonrat there's nothing wrong with a circular dependency here.  I think those methods perfectly validly fit in `Text`.  If at some point you'd like to move them to text utils that's fine, but that seems out-of-scope for this PR.